src/components/bulkConcepts/addBulkConcepts.jsx

```diff
@@ -1,37 +1,88 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import PropTypes from 'prop-types';
+import propTypes from 'prop-types';
 import './style.css';
 import fetchCielConcepts, { addExistingBulkConcepts } from '../../redux/actions/bulkConcepts';
 import BulkConceptList from '../bulkConcepts/bulkConceptList';
+import Paginations from './component/Paginations';
 import Header from '../bulkConcepts/container/Header';
 
 export class AddBulkConcepts extends Component {
   static propTypes = {
-    fetchCielConcepts: PropTypes.func.isRequired,
-    addExistingBulkConcepts: PropTypes.func.isRequired,
-    cielConcepts: PropTypes.arrayOf(PropTypes.shape({
-      id: PropTypes.string,
+    fetchCielConcepts: propTypes.func.isRequired,
+    addExistingBulkConcepts: propTypes.func.isRequired,
+    cielConcepts: propTypes.arrayOf(propTypes.shape({
+      id: propTypes.string,
     })).isRequired,
-    match: PropTypes.shape({
-      params: PropTypes.shape({
-        type: PropTypes.string,
-        typeName: PropTypes.string,
-        collectionName: PropTypes.string,
+    isFetching: propTypes.bool.isRequired,
+    match: propTypes.shape({
+      params: propTypes.shape({
+        type: propTypes.string,
+        typeName: propTypes.string,
+        collectionName: propTypes.string,
       }).isRequired,
     }).isRequired,
-    isFetching: PropTypes.bool.isRequired,
   };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentPage: 1,
+      cielConcepts: [],
+    };
+  }
+
   handleAddAll=() => {
-    const expressions = this.props.cielConcepts.map(concept => concept.url);
-    this.props.addExistingBulkConcepts({ data: { expressions } });
+    const { type, typeName, collectionName } = this.props.match.params;
+    const url = `${type}/${typeName}/collections/${collectionName}/references/`;
+    const data = this.state.cielConcepts;
+    this.props.addExistingBulkConcepts({ url, data: { data: { expressions: data } } });
+    setTimeout(() => {
+      window.history.back();
+    }, 2000);
   }
+
   handleCielClick=() => {
     this.props.fetchCielConcepts();
   }
+
+  handleBack=() => {
+    window.history.back();
+  }
+
+  handleSelect = (e) => {
+    e.preventDefault();
+    const { value } = e.target;
+    if (e.target.checked) {
+      this.setState({
+        cielConcepts: [value, ...this.state.cielConcepts],
+      });
+    } else {
+      const index = this.state.cielConcepts.indexOf(value);
+      this.setState({
+        cielConcepts: this.state.cielConcepts.filter((_, i) => i !== index),
+      });
+    }
+  }
+
+  handleClick = (e) => {
```

Comment:
Is there one click event on this component. More description of the event handler would seem better. What do you think?

Reply:
That's a good Idea. I will fix this.